components/dada-parse/src/parser/items.rs

```diff
@@ -29,10 +29,8 @@ impl<'db> Parser<'db> {
     fn parse_item(&mut self) -> Option<Item> {
         if let Some(class) = self.parse_class() {
             Some(Item::Class(class))
-        } else if let Some(function) = self.parse_function() {
-            Some(Item::Function(function))
         } else {
-            None
+            self.parse_function().map(Item::Function)
```

Comment:
Did this come from clippy? I find this a regression. It's both harder to read and harder to add more cases in the future. I think I'd be inclined to disable whatever clippy lint this is (crate-wide).

Reply:
Yep: https://rust-lang.github.io/rust-clippy/master/index.html#manual_map

```shell
error: manual implementation of `Option::map`
  --> components/dada-parse/src/parser/items.rs:32:16
   |
32 |           } else if let Some(function) = self.parse_function() {
   |  ________________^
33 | |             Some(Item::Function(function))
34 | |         } else {
35 | |             None
36 | |         }
   | |_________^ help: try this: `{ self.parse_function().map(Item::Function) }`
   |
   = note: `-D clippy::manual-map` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_map
```

I agree that this does not help here. The `None` is not only returned because `self.parse_function()` returns `None`, but because all the above options were also `None`. I'll add an `#[allow(...)]` here but I do think this lint can be useful elsewhere.

Reply:
I'm of mixed minds. In general I find that it's often harder to read code that makes extensive use of higher-level helpers like `map`, so sometimes I intentionally write things in a simpler way. On the other hand, I prefer to have fewer customizations where possible, so let's add the "local allow" for now, that seems reasonable.